src/main/java/com/dominik/typer/service/betpersistence/BetService.java

```diff
@@ -33,27 +34,47 @@ public List<Bet> getBetsFromUser(String username) {
         betPersistence.getBetsFromUser(getIdFromUsername(username));
         return betPersistence.getBetsFromUser(getIdFromUsername(username));
     }
-    private void checkIfMatchExists(Integer matchId){
+
+    public List<Bet> getBetsByMatchId(Integer matchId) {
+        checkIfMatchExists(matchId);
+        return betPersistence.getBetsFromMatch(matchId);
+    }
+
+    private void checkIfMatchExists(Integer matchId) {
         if (matchService.getMatch(matchId).isEmpty()) {
             throw new MyAppException("Match does not exist");
         }
     }
+
     private void checkIfUserExistsByUsername(String username) {
         if (userService.getUserWithUsername(username).isEmpty()) {
             throw new MyAppException("User does not exist");
         }
     }
+
     private Integer getIdFromUsername(String username) {
         return userService.getUserWithUsername(username).get().getId();
     }
-    private void checkIfUserHasMoneyForBet(String username, BigDecimal betValue) {
+
+    private void checkIfUserHasMoneyForBet(String username, Double betValue) {
         if (userService.getUserWithUsername(username).get().getBalance().compareTo(betValue) < 0) {
             throw new MyAppException("User does not have enough money");
         }
     }
+
     private void checkIfUserExistsById(Integer id) {
         if (userService.getUser(id).isEmpty()) {
             throw new MyAppException("User does not exist");
         }
     }
+
+    private double checkOddsForBet(Integer id, MatchOutcome betType) {
+        Match match = matchService.getMatch(id).get();
+        Double odds = switch (betType) {
```

Comment:
ostatnio pytałeś mnie czy do klas modelowych można dodać jakąś logikę biznesową (powiedziałem, że raczej nie) - i są wyjątki, i to może się kwalifikować jako wyjątek - można z tego zrobić getter w Match np getOddsByMatchOutcome(MatchOutcome) gdyż nie ma tu jakiejś skomplikowaniej logiki biznesowej a w sumie wszystkie działania są robione wględem modelu - więc jest to dobry przykład wyjątku

Reply:
zmienione